tests/test_whitelist.py

```diff
@@ -27,11 +28,814 @@ def test_load_org_asn(org, asn, mock_db):
+            "Google",
+            json.dumps(["AS15169"]),
+            {"asn": {"number": "AS15169", "asnorg": "Google"}},
+            True,
+        ),
+        (
+            "1.1.1.1",
+            "Cloudflare",
+            json.dumps(["AS13335"]),
+            {"asn": {"number": "AS15169", "asnorg": "Google"}},
+            False,
+        ),
+        ("9.9.9.9", "IBM", json.dumps(["AS36459"]), {}, None),
+        (
+            "9.9.9.9",
+            "IBM",
+            json.dumps(["AS36459"]),
+            {"asn": {"number": "Unknown"}},
+            False,
+        ),
+    ],
+)
+def test_is_ip_asn_in_org_asn(
+    ip, org, org_asn_info, ip_asn_info, expected_result, mock_db
+):
+    whitelist = ModuleFactory().create_whitelist_obj(mock_db)
+    mock_db.get_org_info.return_value = org_asn_info
+    mock_db.get_ip_info.return_value = ip_asn_info
+    result = whitelist.is_ip_asn_in_org_asn(ip, org)
+    assert result == expected_result
+
+
+def test_parse_whitelist(mock_db):
+    whitelist = ModuleFactory().create_whitelist_obj(mock_db)
+    mock_whitelist = {
+        "IPs": json.dumps(
+            {"1.2.3.4": {"from": "src", "what_to_ignore": "both"}}
+        ),
+        "domains": json.dumps(
+            {"example.com": {"from": "dst", "what_to_ignore": "both"}}
+        ),
+        "organizations": json.dumps(
+            {"google": {"from": "both", "what_to_ignore": "both"}}
+        ),
+        "mac": json.dumps(
+            {"b1:b1:b1:c1:c2:c3": {"from": "src", "what_to_ignore": "alerts"}}
+        ),
+    }
+    (
+        whitelisted_IPs,
+        whitelisted_domains,
+        whitelisted_orgs,
+        whitelisted_macs,
+    ) = whitelist.parse_whitelist(mock_whitelist)
+    assert "1.2.3.4" in whitelisted_IPs
+    assert "example.com" in whitelisted_domains
+    assert "google" in whitelisted_orgs
+    assert "b1:b1:b1:c1:c2:c3" in whitelisted_macs
+
+
+def test_get_all_whitelist(mock_db):
+    whitelist = ModuleFactory().create_whitelist_obj(mock_db)
+    mock_db.get_all_whitelist.return_value = {
+        "IPs": json.dumps(
+            {"1.2.3.4": {"from": "src", "what_to_ignore": "both"}}
+        ),
+        "domains": json.dumps(
+            {"example.com": {"from": "dst", "what_to_ignore": "both"}}
+        ),
+        "organizations": json.dumps(
+            {"google": {"from": "both", "what_to_ignore": "both"}}
+        ),
+        "mac": json.dumps(
+            {"b1:b1:b1:c1:c2:c3": {"from": "src", "what_to_ignore": "alerts"}}
+        ),
+    }
+    all_whitelist = whitelist.get_all_whitelist()
+    assert all_whitelist is not None
+    assert "IPs" in all_whitelist
+    assert "domains" in all_whitelist
+    assert "organizations" in all_whitelist
+    assert "mac" in all_whitelist
+
+
+@pytest.mark.parametrize(
+    "flow_data, whitelist_data, expected_result",
+    [
+        (  # testing_is_whitelisted_flow_with_whitelisted_organization_but_ip_or_domain_not_whitelisted
+            MagicMock(
```

Comment:
hey you can use spaces as word separators in comments instead of _ :))

Reply:
That's just a mistake I made while typing quickly. Sorry about that.